warehouse/query-core/src/main/java/datawave/query/jexl/visitors/FixNegativeNumbersVisitor.java

```diff
@@ -19,23 +20,34 @@ public static ASTJexlScript fix(JexlNode root) {
     public Object visit(ASTUnaryMinusNode astumn, Object data) {
         if (astumn.jjtGetNumChildren() == 1 && astumn.jjtGetChild(0) instanceof ASTNumberLiteral) {
             ASTNumberLiteral node = (ASTNumberLiteral) astumn.jjtGetChild(0);
-            ASTNumberLiteral newNode = new ASTNumberLiteral(ParserTreeConstants.JJTNUMBERLITERAL);
-            String value = "-" + node.image;
-            newNode.image = value;
-            newNode.jjtSetParent(node.jjtGetParent());
-            if (JexlNodeFactory.NATURAL_NUMBERS.contains(node.getLiteralClass())) {
-                newNode.setNatural(value);
-            } else if (JexlNodeFactory.REAL_NUMBERS.contains(node.getLiteralClass())) {
-                newNode.setReal(value);
-            } else {
+            ASTNumberLiteral newNode = JexlNodes.makeNumberLiteral();
+            Number value = negate(node.getLiteral());
+            if (!JexlNodes.setLiteral(newNode, value)) {
                 throw new IllegalArgumentException("Could not ascertain type of ASTNumberLiteral: " + node);
             }
-            newNode.jjtSetValue(value);
+            newNode.jjtSetParent(node.jjtGetParent());
             return newNode;
         } else {
             return super.visit(astumn, data);
         }
+    }
 
+    private Number negate(Number number) {
+        Number negated = null;
+        if (number instanceof Integer) {
+            negated = -number.intValue();
+        } else if (number instanceof Long) {
+            negated = -number.longValue();
+        } else if (number instanceof BigInteger) {
+            negated = ((BigInteger) number).negate();
+        } else if (number instanceof Float) {
+            negated = -number.floatValue();
+        } else if (number instanceof Double) {
+            negated = -number.doubleValue();
+        } else if (number instanceof BigDecimal) {
+            negated = ((BigDecimal) number).negate();
+        }
```

Comment:
perhaps the default action should be negated = -number.doubleValue();

Reply:
The previous code throws an illegal argument exception if the value isn't a known type (byte, short, integer, long, biginteger, float, double, bigdecimal), so I updated the code to match that behavior.